test/spark.js

```diff
@@ -159,20 +159,34 @@ test('fetchCAR fails with statusCode=702 (protocol is not tcp)', async () => {
   assertEquals(stats.statusCode, 702, 'stats.statusCode')
 })
 
-test('fetchCAR fails with statusCode=703 (scheme is not http/https)', async () => {
+test('fetchCAR fails with statusCode=703 (scheme is not http/https) - multiaddr without http-path', async () => {
   const spark = new Spark()
   const stats = newStats()
   await spark.fetchCAR('http', '/ip4/1.2.3.4/tcp/80/ldap', KNOWN_CID, stats)
   assertEquals(stats.statusCode, 703, 'stats.statusCode')
 })
 
+test('fetchCAR fails with statusCode=703 (scheme is not http/https) - multiaddr with http-path', async () => {
```

Comment:
```suggestion
test('fetchCAR fails with statusCode=703 (scheme is not supported) - multiaddr with http-path', async () => {
```

To me it wasn't clear immediately why "not http/https" caused an error

Reply:
This has to do with the [change](https://github.com/CheckerNetwork/spark-checker/pull/116#discussion_r1965542944) that @bajtos mentioned. We can either put the scheme check first or the check for the length of the rest. If we put the check for the scheme first then this test will fail with the error message that the scheme is not supported. The scheme is in this case `tcp` which is not `http` or `https`. We could make additional checks in the conditional branch for `http-path` but since miro mentioned that the function is already [too complex](https://github.com/CheckerNetwork/spark-checker/pull/116#pullrequestreview-2633133366) I did not want to add complexity by adding this workaround.

Reply:
Sorry my comment was only about the test name, if we rename it like this it's more clear to me what the intend of the test is

Reply:
Sure :)